### PR TITLE
docker/ci: Add debugging info

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@lizan i think we should add this first before making too much changes in the docker ci file as its a bit complex and hard to track

one thing running this reveals straight away is that (at least with the context i was testing) the release images are pushed  (afaict needlessly) a second time - not sure if we can just remove - windows is a factor

File: ci/docker_ci.sh [hunk 1]
```diff
@@ -136,7 +154,9 @@ if [[ "${AZP_BRANCH}" != "${MAIN_BRANCH}" ]] &&
   exit 0
 fi
 
-docker login -u "$DOCKERHUB_USERNAME" -p "$DOCKERHUB_PASSWORD"
+if [[ -z "$DOCKER_CI_DEBUG" || -n "$DOCKER_CI_DEBUG_LOGIN" ]]; then
```
Comment (phlax) [inline ci/docker_ci.sh:157]:
```suggestion
if [[ -z "$DOCKER_CI_DEBUG" ]]; then
```

this is not necessary if you want to debug with a real build to a fake repo with fake context - just dont set the debug var

File: ci/docker_ci.sh [hunk 2]
```diff
@@ -85,9 +113,19 @@ push_images() {
...
+    docker push ${BUILD_TAG}"
+  if [[ -n "$DOCKER_CI_DRYRUN" ]]; then
+      echo ""
+      return
+  fi
+  echo "..."
   # docker buildx doesn't do push with default builder
   docker "${BUILD_COMMAND[@]}" --platform "${PLATFORM}" "${args[@]}" -t "${BUILD_TAG}" . --push ||
     docker push "${BUILD_TAG}"
+  echo ""
```
Comment (wbpcode) [inline ci/docker_ci.sh:128]: is this necessary?

Comment (phlax) [reply ci/docker_ci.sh:128]: it makes it easier to parse the output - ie it separates the blocks

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**